### DISCUSSION TITLE
ModelZoo

### DISCUSSION THREAD

**Discussion by bobo0810** [hpcaitech/ColossalAI#2843] [Community | Q&A] [answered]
是否计划提供官方模型库，以便直接启用Tensor、流水线等并行策略。

**Comment by bobo0810** [comment]
注意到近期的有关Autoparallel的更新，是否意味着  理论上任何模型经过Autoparallel简单包装即可直接启用colossalai的各类并行策略呢？

**Reply by FrankLeeeee** [reply]
理想情况是这样的，目前还在积极开发中，目前的目标是覆盖常见的模型比如bert/gpt/vit等

**Reply by bobo0810** [reply]
> 理想情况是这样的，目前还在积极开发中，目前的目标是覆盖常见的模型比如bert/gpt/vit等

期待🎉     AutoParallel预计何时出教程中呢？

**Reply by FrankLeeeee** [reply]
目前已经有一个教程在
https://github.com/hpcaitech/ColossalAI/tree/main/docs/source/en/Colossal-Auto

但是还没有正式发布，可能这两周会先发一下experimental feature+简单的demo

**Reply by YuliangLiu0306** [reply]
目前可以根据[example](https://github.com/hpcaitech/ColossalAI/tree/main/examples/language/gpt/experiments/auto_parallel)和一些[文档](https://github.com/hpcaitech/ColossalAI/blob/main/colossalai/auto_parallel/README.md)来了解自动并行的使用方法。我们后续会在功能完善之后，出更详细的使用教程

**Comment by FrankLeeeee** [comment, marked as answer]
> 是否计划提供官方模型库，以便直接启用Tensor、流水线等并行策略。

我们在考虑直接在huggingface的模型库上进行并行，目前还在设计中。

**Reply by bobo0810** [reply]
> 我们在考虑直接在huggingface的模型库上进行并行，目前还在设计中。

目前timm模型库提供约800+预训练模型，是事实上的基础模型库。是否在计划内呢？

**Reply by FrankLeeeee** [reply]
不太可能立刻支持这么多模型，会有corner case处理比较困难，目前还是以主流大模型为主，timm和huggingface里很多小模型其实没有必要用自动并行，gemini足够了

**Reply by bobo0810** [reply]
> 不太可能立刻支持这么多模型，会有corner case处理比较困难，目前还是以主流大模型为主，timm和huggingface里很多小模型其实没有必要用自动并行，gemini足够了

重点支持vit、swinv2等Transformer类型即可。期待

**Reply by FrankLeeeee** [reply]
That's right. 感谢关注！:)
